llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2301,9 +2300,11 @@ class VPReductionRecipe : public VPRecipeWithIRFlags {
              VPSlotTracker &SlotTracker) const override;
 #endif
 
-  /// Return the recurrence decriptor for the in-loop reduction.
-  const RecurrenceDescriptor &getRecurrenceDescriptor() const {
-    return RdxDesc;
+  /// Return the recurrence kind for the in-loop reduction.
+  RecurKind getRecurrenceKind() const { return RdxKind; }
+  /// Return the opcode for the recurrence for the in-loop reduction.
+  unsigned getOpcode() const {
+    return RecurrenceDescriptor::getOpcode(RdxKind);
```

Comment:
If we don't expect `RdxKind` to change during the class lifetime, is it worth just caching the opcode in the class at the time of construction?

Reply:
I've inlined it in 23bcb59dc2603d837e285ab8c8f34cde7cc932b9 which should be more similar to the existing code